plutus-core/executables/debugger/Main.hs

```diff
@@ -207,20 +181,65 @@ driverThread driverMailbox brickMailbox prog = do
         D.StepF prevState k  -> cekMToIO (D.handleStep prevState) >>= k
         D.InputF k           -> handleInput >>= k
         D.LogF text k        -> handleLog text >> k
-        D.UpdateClientF ds k -> handleUpdate ds >> k -- TODO: implement
+        D.UpdateClientF ds k -> handleUpdate ds >> k
       where
         handleInput = takeMVar driverMailbox
         handleUpdate = B.writeBChan brickMailbox . UpdateClientEvent
         handleLog = B.writeBChan brickMailbox . LogEvent
 
-unDeBruijnProgram ::
-    UPLC.Program UPLC.NamedDeBruijn DefaultUni DefaultFun () ->
-    IO (UPLC.Program UPLC.Name DefaultUni DefaultFun ())
-unDeBruijnProgram p = do
-    either (fail . show) pure
-        . PLC.runQuote
-        . runExceptT @UPLC.FreeVariableError
-        $ traverseOf UPLC.progTerm UPLC.unDeBruijnTerm p
+-- | Read uplc code in a given format
+--
+--  Adaptation of `Common.getProgram`
```

Comment:
I _really_ want to do that thing where we add "text envelopes" for serialized programs that tell us the type. This stuff is such a pain otherwise.